src/plugins/displayLayout/DisplayLayoutToolbar.js

```diff
@@ -623,6 +623,33 @@ define(['lodash'], function (_) {
                     }
                 }
 
+                function getToggleGridButton(selection, selectionPath) {
+                    const ICON_GRID_SHOW = 'icon-grid-show';
+                    const ICON_GRID_HIDE = 'icon-grid-hide';
+
+                    let displayLayoutContext;
+
+                    if (selection.length === 1 && selectionPath === undefined) {
+                        displayLayoutContext = selection[0][0].context;
+                    } else {
+                        displayLayoutContext = selectionPath[1].context;
+                    }
+
+                    return {
+                        control: "button",
+                        domainObject: displayLayoutContext.item,
+                        icon: ICON_GRID_SHOW,
+                        method: function () {
+                            this.icon = this.icon === ICON_GRID_SHOW
```

Comment:
Lets move switching the icon to after the toggleGrid call, that way if toggle grid fails, then the icon will not change.